core/src/lib.rs

```diff
@@ -291,12 +297,60 @@ impl Schema {
     pub fn identifies_attribute(&self, x: &NamespacedKeyword) -> bool {
         self.get_entid(x).map(|e| self.is_attribute(e)).unwrap_or(false)
     }
+
+    /// Returns an edn::Value::Vector representation of the schema map.
+    pub fn to_edn_value(&self) -> edn::Value {
+        let mut all_values = Vec::new();
+        for (entid, attribute) in &self.schema_map {
+            let mut attribute_map: BTreeMap<edn::Value, edn::Value> = BTreeMap::default();
+            let some_ident = self.get_ident(entid.clone());
+            if some_ident.is_some() {
+                let ident = some_ident.unwrap();
+                attribute_map.insert(values::DB_IDENT.clone(), edn::Value::NamespacedKeyword(ident.clone()));
+            } else {
+                attribute_map.insert(values::DB_IDENT.clone(), edn::Value::Keyword(Keyword::new(entid.to_string())));
+            }
+
+            let value_type = format!("{:?}", attribute.value_type);
```

Comment:
You're relying on the debug format (`{:?}`) of an enum here.

Given a `ValueType`, you're trying to get a `NamespacedKeyword`. There are two ways to do that:

- Go from the `ValueType` to an entid, and look up the ident.
- Go directly from a `ValueType` to a `NamespacedKeyword`.

Either way, break that out as a helper…